apps/jsonrpc2/lib/jsonrpc2/spec_handler.ex

```diff
@@ -1,5 +1,127 @@
 defmodule JSONRPC2.SpecHandler do
   use JSONRPC2.Server.Handler
 
-  def handle_request("web3_clientVersion", _), do: Application.get_env(:jsonrpc2, :mana_version)
+  alias Blockchain.Block
+  alias Blockchain.Blocktree
+  alias ExthCrypto.Hash.Keccak
+  alias ExthCrypto.Math
+  alias ExWire.PeerSupervisor
+  alias ExWire.Sync
+  # TODO split methods into separate modules per topic
+  # web3 Methods
+
+  def handle_request("web3_clientVersion", _),
+    do: "#{Application.get_env(:jsonrpc2, :mana_version)}"
+
+  def handle_request("web3_sha3", [param = "0x" <> _]) do
+    "0x" <>
+      (param
+       |> Math.hex_to_bin()
+       |> Keccak.kec()
+       |> Math.bin_to_hex())
+  rescue
+    _ ->
+      {:error, :invalid_params}
```

Comment:
maybe we can move this to somewhere (module attribute or separate module) to reuse in other methods?

Reply:
Which part exactly?

Reply:
`:invalid_params` atom or `{:error, :invalid_params}` tuple

Reply:
Ok, yeah, I know what you mean now. I'll do that once I get a couple of more examples cleared out so that I understand the problem more.